src/lib/spinel/radio_spinel.hpp

```diff
@@ -46,6 +46,103 @@
 namespace ot {
 namespace Spinel {
 
+/**
+ * Maximum number of Spinel Interface IDs.
+ *
+ */
+#ifdef OPENTHREAD_CONFIG_MULTIPAN_RCP_ENABLE
+static constexpr uint8_t kSpinelHeaderMaxNumIid = 4;
+#else
+static constexpr uint8_t kSpinelHeaderMaxNumIid = 1;
+#endif
+
+struct RadioSpinelCallbacks
```

Comment:
Curious, why do we need a struct for spinel callbacks now?

Reply:
While we focus here on the RCP side, we want to cover different possibilities for host implementations. Callbacks needs to be used to cover the case when all interfaces operate under the same process with multiple instances of RadioSpinel class. We may want to use different API for different interfaces (for example if not using OT for the second interface)